tests/common/helpers/snmp_helpers.py

```diff
@@ -14,35 +14,46 @@
 global_snmp_facts = {}
 
 
+def is_snmp_subagent_running(duthost):
+    cmd = "docker exec snmp supervisorctl status snmp-subagent"
+    output = duthost.shell(cmd)
+    if "RUNNING" in output["stdout"]:
+        logger.info("SNMP Sub-Agent is Running")
+        return True
+    logger.info("SNMP Sub-Agent is Not Running")
+    return False
+
+
 def _get_snmp_facts(localhost, host, version, community, is_dell, include_swap, module_ignore_errors):
     snmp_facts = localhost.snmp_facts(host=host, version=version, community=community, is_dell=is_dell,
                                       module_ignore_errors=module_ignore_errors, include_swap=include_swap)
     return snmp_facts
 
 
-def _update_snmp_facts(localhost, host, version, community, is_dell, include_swap):
+def _update_snmp_facts(localhost, host, version, community, is_dell, include_swap, duthost):
     global global_snmp_facts
 
     try:
+        snmp_subagent_running = is_snmp_subagent_running(duthost)
```

Comment:
@agadia-cisco 
If we can get expected output from snmp_facts means snmp-subagent is ready already?
If the readiness of '_get_snmp_facts' implicitly include the readiness of snmp-subagent,
We already have the logic of wait_for _update_snmp_facts ?

Or do you mean even if the "Timeout issues" or "Missing Information" in snmp_facts output, except RunAnsibleModuleFail as e: can't catch them, so just return an empty or wrong snmp_facts?

Reply:
_"If we can get expected output from snmp_facts means snmp-subagent is ready already?"_

No, earlier we were just calling `snmp_facts` (which doesn't have a pre-requisite for `snmp-subagent` to be up); thereby if there was a delay in `snmp-subagent` to be up & `snmp_facts` was already called, then we observed "Timeout Issues" mostly & in some scenarios TCs failed due to missing keys returned in `snmp_facts` output & those keys being used in TCs.

Reply:
_"If the readiness of `_get_snmp_facts` implicitly include the readiness of `snmp-subagent`, We already have the logic of `wait_for _update_snmp_facts` ?"_

`get_snmp_facts` is calling ->
- `_get_snmp_facts` directly when `wait` is False 
- `_update_snmp_facts` directly when `wait` is True, which calls `_get_snmp_facts`

reason, I didn't add `is_snmp_subagent_running` check in `_get_snmp_facts` is because as per `get_snmp_facts` function design, we don't want to wait if the `wait` is False, & so I added the check with `_update_snmp_facts` because we are calling it when `wait` is True

I also checked all the TCs in `sonic-mgmt` wherever `get_snmp_facts` was called & almost all of them had been passing `wait` as True thereby `_update_snmp_facts` being called.